src/debug/debugSession.ts

```diff
@@ -104,6 +105,97 @@ export class DebugSession implements IDebugSession {
         });
     }
 
+    /**
+     * In attach mode, the user should choose the running pod first, then the debugger will attach to it.
+     *
+     * @param workspaceFolder the active workspace folder.
+     * @param pod the target pod name.
+     */
+    public async attach(workspaceFolder: vscode.WorkspaceFolder, pod?: string): Promise<void> {
+        if (!workspaceFolder) {
+            return;
+        }
+
+        // Select the image type to attach.
+        this.debugProvider = await providerRegistry.getDebugProvider();
+        if (!this.debugProvider) {
+            return;
+        } else if (!await this.debugProvider.isDebuggerInstalled()) {
+            return;
+        }
+
+        // Select the target pod to attach.
+        let targetPod, targetContainer, containers = [];
+        if (pod) {
+            targetPod = pod;
+            const shellResult = await this.kubectl.invokeAsync(`get pod/${pod} -o json`);
+            if (shellResult.code !== 0) {
+                vscode.window.showErrorMessage(shellResult.stderr);
+                return;
+            }
+            containers = JSON.parse(shellResult.stdout.trim()).spec.containers;
+        } else {
+            const shellResult = await this.kubectl.invokeAsync("get pods -o json");
+            if (shellResult.code !== 0) {
+                vscode.window.showErrorMessage(shellResult.stderr);
+                return;
+            }
+            const podObj = JSON.parse(shellResult.stdout.trim());
+            const podPickItems: vscode.QuickPickItem[] = podObj.items.map((pod) => {
+                return {
+                    label: `${pod.metadata.name} (${pod.spec.nodeName})`,
+                    description: "pod",
+                    name: pod.metadata.name,
+                    containers: pod.spec.containers
+                };
+            });
+            const selectedPod = await vscode.window.showQuickPick(podPickItems, { placeHolder: `Please select a pod to attach` });
+            if (!selectedPod) {
+                return;
+            }
+            targetPod = (<any> selectedPod).name;
+            containers = (<any> selectedPod).containers;
+        }
+
+        // Select the target container to attach.
+        if (containers.length > 1) {

```

Comment:
I thinks we'd better provide pod+container options instead of choosing pod and choosing container

Reply:
If flatten with pod+container in a row, the row will become too long and not good for small screen.